### ISSUE TITLE
SC2257 should cover additional state-changing expansions

### ISSUE THREAD

**Issue by dimo414** [koalaman/shellcheck#2015]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2257
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
date >"${word:=wow}.txt"
```

#### Here's what shellcheck currently says:

No issues detected!

#### Here's what I wanted or expected to see:

This is an instance of SC2257; the `:=` expansion will not update the enclosing shell's state (unless the command is a builtin/function).
